browser/brave_wallet/android/brave_wallet_native_worker.cc

```diff
@@ -48,80 +70,83 @@ void BraveWalletNativeWorker::Destroy(
 
 base::android::ScopedJavaLocalRef<jstring>
 BraveWalletNativeWorker::GetRecoveryWords(JNIEnv* env) {
-  auto* keyring_controller = GetBraveWalletService()->keyring_controller();
-  keyring_controller->GetMnemonicForDefaultKeyring();
-  return base::android::ConvertUTF8ToJavaString(
-      env, keyring_controller->GetMnemonicForDefaultKeyring());
+  EnsureConnected();
+
+  auto callback = base::BindOnce([](const std::string& mnemonic) {});
+  keyring_controller_->GetMnemonicForDefaultKeyring(std::move(callback));
+
+  // TODO(bridiver) - convert this to callback
```

Comment:
same as ios these will be fixed in the follow-up. The current handling for callbacks in JNI is subject to multiple race conditions so not much point in fixing this now and @SergeyZhukovsky agrees